### DISCUSSION TITLE
Avoid null keys with enabled option

### DISCUSSION THREAD

**Discussion by BillySemana** [TanStack/query#2982] [Q&A] [answered]
Hello everyone 👋 ,

I don't know if it's the expected behaviour but I have "null" keys in my cache even when using the `enabled` option.

I'm using a hook for fetching a collaborator by id
```typescript
const useCollaborator(id?: number) {
  return useQuery(["collaborators", id], fetchCollaborator, {
      enabled: !!id
  })
}

```

I still have a cache with the following keys `["collaborators", null]`
I was expecting to not having a cache at all when `id` is undefined.

Is there a way to avoid that ?

**Comment by TkDodo** [comment, marked as answer]
The query should show up as `disabled` in the devtools. There is no way to avoid that, because the query has been created, even if it's disabled. You could pre-fill it with `initialData`, and you can trigger a fetch with `refetch` returned from `useQuery` even if the query is disabled. Does the cache entry bother you?

**Reply by mossen** [reply]
In my case what bothers me is I cannot get the data if the cache entry is there, compare following entries:

```js
const versionData = queryClient.getQueryData(
    [QueryKeys.VERSION],
    { exact: false },
  );
  const projectConfigData = queryClient.getQueryData(
    [QueryKeys.PROJECT_CONFIG],
    { exact: false },
  );
```

<img width="400" alt="Screenshot 2023-01-23 at 1 34 34 pm" src="https://user-images.githubusercontent.com/14992360/213956908-5c345bc0-9d42-4c97-b87e-595cf8c02255.png">

`projectConfigData` is undefined as the key confuses the filter, I need to get the `["project-config","rb.gcf.project-config.csv"]` but it works for `versionData`
